site/docs/v1/tech/apis/hosted-backend.adoc

```diff
@@ -0,0 +1,400 @@
+
+[field]#state# [type]#[String]# [optional]#Optional#::
+The state that was provided on the Authorization request.
+This parameter will be omitted if the [field]#state# request parameter was not provided.
+
+[field]#userState# [type]#[String]# [optional]#Optional#::
+The FusionAuth user state.
++
+The possible values are:
++
+* `Authenticated` - The user is successfully authenticated for the requested application.
+* `AuthenticatedNotRegistered` - The user is successfully authenticated, but not registered for the requested application.
+
+.Example Request URL
+[source,url]
+----
+https://auth.yourdomain.io/app/callback?code=wJfjafZLvo_KH5-D4r-3YwMmStN3yHoZDGmBivjioz0&locale=en&state=eyJjIjoiODVhMDM4NjctZGNjZi00ODgyLWFkZGUtMWE3&userState=Authenticated&client_id=297ca84b-69a9-4508-8649-97644e1d0b3d&tenantId=e707be45-afa8-4881-9efb-4be7288395d2
+----
+
+=== Response
+
+A successful response should set cookies and return a `302` redirect to the `redirect_uri` specified in the initial request. Other status codes indicate an error.
+
+[cols="1a,9a"]
+.Response Codes
+|===
+|Code |Description
+
+|200
+|This is likely an error case. The route will serve up an error page with html and details on what went wrong.
+
+|302
+|A successful request will redirect the user to `redirect_uri` specified in the request or the default Authorized Redirect URL configured for the application.
+
+|500
+|There was a FusionAuth internal error. A stack trace is provided and logged in the FusionAuth log files.
+|===
+
+== Refresh
+
+This endpoint will extract the `app.rt` cookie if present and use it to make a `refresh_token` request from `/oauth/token`. If successful a new set of cookies will be set on the response the will continue to allow access to the application.
+
+This is an api that should be made from the client application and the browser should *NOT* be redirected to this endpoint.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#jwt-authentication[icon:id-badge[type=fas]] Refresh the current session cookies.
+
+[.endpoint]
+.URI
+--
+[method]#POST# [uri]#/app/refresh/{clientId}#
+--
+
+==== Request Parameters
+
+[.api]
+[field]#client_id# [type]#[UUID]# [required]#Required#::
+The top-level client Id for your application.
+
+.Example Request URL
+[source,url]
+----
+https://auth.yourdomain.io/app/refresh/297ca84b-69a9-4508-8649-97644e1d0b3d
+----
+
+=== Response
+
+A successful response should set cookies and return a `200`.
+
+[cols="1a,9a"]
+.Response Codes
+|===
+|Code |Description
+
+|200
+|This is likely an error case. The route will serve up an error page with html and details on what went wrong.
+
+|400
+|The request was not successful. The client may need to reauthorize.
+
+|500
+|There was a FusionAuth internal error. A stack trace is provided and logged in the FusionAuth log files.
+|===
+
```

Comment:
We should document the response cookies like we do here: https://fusionauth.io/docs/v1/tech/apis/login#authenticate-a-user